java/developer-guide/working-with-outlook-items/working-with-recurrences/_index.md

```diff
@@ -0,0 +1,527 @@
+
+
+
+~~~Java
+// For complete examples and data files, please go to https://github.com/aspose-email/Aspose.Email-for-Java
+// Set the Daily recurrence
+MapiCalendarDailyRecurrencePattern record = new MapiCalendarDailyRecurrencePattern();
+
+record.setPatternType(MapiCalendarRecurrencePatternType.Day);
+record.setPeriod(1);
+record.setWeekStartDay(DayOfWeek.Sunday);
+record.setEndType(MapiCalendarRecurrenceEndType.EndAfterNOccurrences);
+record.setOccurrenceCount(5);
+task.setRecurrence(record);
+~~~
+### **Daily Recurrences: EndAfterDate Recurrance Type**
+"End By" option in the Mapi Task is achieved by setting the OccurrenceCount property calculated by the getOccurrenceCount() function. This function takes start date , end date and RRULE string.
+#### **Daily Recurrences: Setting the Every Day value**
+The following code snippet shows you how to set the period value to 1 and INTERVAL value to 1 in the RRULE string as well.
+
+
+
+~~~Java
+// For complete examples and data files, please go to https://github.com/aspose-email/Aspose.Email-for-Java
+// Set the Daily recurrence
+MapiCalendarDailyRecurrencePattern record = new MapiCalendarDailyRecurrencePattern();
+
+record.setPatternType(MapiCalendarRecurrencePatternType.Day);
+record.setPeriod(1);
+record.setEndType(MapiCalendarRecurrenceEndType.EndAfterDate);
+record.setOccurrenceCount(getOccurrenceCount(startDate, endByDate, "FREQ=DAILY;INTERVAL=1"));
+record.setEndDate(endByDate);
+~~~
+
+
+
+Every Day value can be set to any appropriate value as shown in the following example:
+
+
+
+~~~Java
+// For complete examples and data files, please go to https://github.com/aspose-email/Aspose.Email-for-Java
+// Set the Daily recurrence
+MapiCalendarDailyRecurrencePattern record = new MapiCalendarDailyRecurrencePattern();
+
+record.setPatternType(MapiCalendarRecurrencePatternType.Day);
+record.setPeriod(2);
+record.setEndType(MapiCalendarRecurrenceEndType.EndAfterDate);
+record.setOccurrenceCount(getOccurrenceCount(startDate, endByDate, "FREQ=DAILY;INTERVAL=2"));
+~~~
+### **Daily Recurrences: NeverEnd Recurrance Type**
+End type can be set by using MapiCalendarRecurrenceEndType.NeverEnd. Period or INTERVAL can be set to required value say 1 in the following example.
+
+
+
+~~~Java
+// For complete examples and data files, please go to https://github.com/aspose-email/Aspose.Email-for-Java
+// Set the Daily recurrence
+MapiCalendarDailyRecurrencePattern record = new MapiCalendarDailyRecurrencePattern();
+
+record.setPatternType(MapiCalendarRecurrencePatternType.Day);
+record.setPeriod(1);
+record.setEndType(MapiCalendarRecurrenceEndType.NeverEnd);
+~~~
+## **Working with Weekly Recurrences**
+Aspose.Email provides rich features for creation of weekly recurrences using MapiCalendarWeeklyRecurrencePattern. Three different Mapi calendar recurrence end types can be used including EndAfterNOccurrences, EndAfterDate and NeverEnd. This section demonstrates the creation of different weekly recurrence patterns.
```

Comment:
`Aspose.Email provides rich features for the creation of weekly recurrences using *MapiCalendarWeeklyRecurrencePattern*. Three different Mapi calendar recurrence end types can be used including *EndAfterNOccurrences*, *EndAfterDate* and *NeverEnd*. This section demonstrates the creation of different weekly recurrence patterns.`